code-challenges.js

```diff
@@ -7,17 +7,48 @@ var secretCodeWord1 = "lackadaisical"
 var secretCodeWord2 = "gobbledygook"
 // Expected output: "g0bbl3dyg00k"
 
+const code = (string) => {
+    //Returns the string argument after splitting it into an array, then mapping through each value(char). If the value is an 'a', 'e', 'i', or 'o', it will replace that letter with a special number assigned each respective letter. Otherwise it will return the normal character and then join the entire array with the replaced values back into a string.
+    return string.split("").map(value => value === 'a' ? 4 : value === 'e' ? 3 : value === 'i' ? 1 : value === 'o' ? 0 : value).join("")
 
-
+    //Refactored this code into the one above to make it look more simplified.
+    // return string.split("").map(value => {
+    //     if (value === 'a') {
+    //         return 4
+    //     }
+    //     else if (value === 'e') {
+    //         return 3
+    //     }
+    //     else if (value === 'i') {
+    //         return 1
+    //     }
+    //     else if (value === 'o') {
+    //         return 0
+    //     }
+    //     else {
+    //         return value
+    //     }
+    // }).join("")
+}
```

Comment:
nice work! I like that both attempts are visible here :) I might just break them into two different functions for readability perhaps?